provider/alibabacloud/alibaba_cloud.go

```diff
@@ -476,9 +475,15 @@ func (p *AlibabaCloudProvider) applyChangesForDNS(changes *plan.Changes) error {
 
 	recordMap := p.groupRecords(records)
 
-	p.createRecords(changes.Create)
+	hostedZoneDomains, err := p.getDomainList()
+
```

Comment:
```suggestion
```

Reply:
sorry, i'm not sure what you mean by this suggested change. the p.createRecords function call is not removed, first we resolve the hostedZoneDomains and pass it to p.createRecords and p.updateRecords (instead of resolving hostedZoneDomains twice)

Reply:
The blank line is unnecessary. The error check should be immediately after the call that returns `err`.

Reply:
done